tests/integration/test_requirements_validation.py

```
@@ -0,0 +1,209 @@
+        assert 'pyyaml' in requirements_dev_content.lower() or \
+               'PyYAML' in requirements_dev_content
+    
+    def test_pyyaml_has_version_specifier(self, requirements_dev_content):
+        """
+        Check that a PyYAML entry in requirements-dev.txt includes a version specifier.
+        
+        Searches the given requirements file content for a line mentioning PyYAML and verifies that the line contains one of the recognised version operators: >=, ==, ~=, <=, > or <.
+        """
+        lines = requirements_dev_content.split('\n')
+        # Ignore commented lines so we don't pick up commented-out examples
+        pyyaml_line = next((l for l in lines if 'pyyaml' in l.lower() and not l.strip().startswith('#')), None)
```

Ruff (0.14.8)

Ambiguous variable name: `l`

(E741)